x-pack/plugins/triggers_actions_ui/public/application/hooks/use_load_tags_query.ts

```diff
@@ -35,21 +58,48 @@ export function useLoadTagsQuery(props: UseLoadTagsQueryProps) {
     );
   };
 
-  const { refetch, data } = useQuery({
-    queryKey: [
-      'loadRuleTags',
-      {
-        refresh: refresh?.toDateString(),
-      },
-    ],
-    queryFn,
-    onError: onErrorFn,
-    enabled,
-    refetchOnWindowFocus: false,
-  });
+  const getNextPageParam = (lastPage: GetRuleTagsResponse) => {
+    const totalPages = Math.max(1, Math.ceil(lastPage.total / lastPage.perPage));
+    if (totalPages === lastPage.page) {
+      return;
+    }
+    return {
+      ...lastPage,
+      page: lastPage.page + 1,
+    };
+  };
+
+  const { refetch, data, fetchNextPage, isLoading, isFetching, hasNextPage, isFetchingNextPage } =
+    useInfiniteQuery({
+      queryKey: [
+        'loadRuleTags',
+        search,
+        perPage,
+        page,
+        {
+          refresh: refresh?.toISOString(),
+        },
+      ],
+      queryFn,
+      onError: onErrorFn,
+      enabled,
+      getNextPageParam,
+      refetchOnWindowFocus: false,
+    });
+
+  const tags = useMemo(() => {
+    return (
+      data?.pages.reduce<string[]>((result, current) => {
+        return [...result, ...current.data];
```

Comment:
you are not scared of duplicate here;
also I think it is better to do that 
```ts
result.push(...current.data);
return result;
```

Reply:
using `concat` instead to avoid all spreads.